src/main/scala/mesosphere/marathon/api/v2/QueueResource.scala

```diff
@@ -37,9 +41,10 @@ class QueueResource @Inject() (
   def resetDelay(
     @PathParam("appId") id: String,
     @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+    //    import scala.concurrent.ExecutionContext.Implicits.global
     val appId = id.toRootPath
-    val maybeApps = result(launchQueue.list)
-    val maybeApp = maybeApps.find(_.runSpec.id == appId).map(_.runSpec)
+    val appScheduled = result(instanceTracker.specInstances(appId)).exists(_.isScheduled)
+    val maybeApp = if (appScheduled) groupManager.runSpec(appId) else None
```

Comment:
You are right. We should verify that the app we retrieve has the same version as the scheduled app. However, currently we have a weird race. The run spec version of a scheduled app can be overridden.

Reply:
overriden where? I still feel it's safer to use the appScheduled version if it's not fundamentally incorrect (and if that is we have probably problems, because that would be weird :))